src/coord/src/coord/arrangement_state.rs

```diff
@@ -104,6 +104,10 @@ pub struct Frontiers<T: Timestamp> {
     /// The most recent frontier for new data.
     /// All further changes will be in advance of this bound.
     pub upper: MutableAntichain<T>,
+    /// The most recent frontier for durable data.
+    /// All data at times in advance of this frontier have not yet been
+    /// durably persisted and may not be replayable across restarts.
+    pub durability: Antichain<T>,
```

Comment:
The coordinator tracks this on an arrangement-by-arrangement basis? Would it make sense to track it on a source basis, rather than continually updating this for all intermediate arrangements?